src/content/docs/en/recipes/build-custom-img-component.mdx

```diff
@@ -0,0 +1,111 @@
+---
+title: Build a custom image component 
+description: Learn how to build a custom image component that supports media queries using the getImage function
+i18nReady: true
+type: recipe
+---
+
+Astro provides a couple of built in components that you can use to display and optimize your images.  The `<Picture>` component allows you to display responsive images and work with different formats and sizes. The `<Image>` componet will optimze your images and allow you to pass in different formats and quality properties.
+
+But there will be times where you will need to have more customizable options for your images that the `<Picture>` and `<Image>` components currently do not support. 
+
+In this recipe, you will learn how use the `getImage()` function to create your own custom image component that supports the use of media queries/art direction.  The purpose of the `getImage()` function is to create images for use outside of direct embedding in HTML.
+
+## Recipe
+
+1. Create a new Astro component and import the `getImage()` function
+
+    ```astro title="src/components/MyCustomImageComponent.astro"
+    ---
+     import { getImage } from "astro:assets";
+    ---
+
+    ```
+
+2. Destructure the `props` from `Astro.props`. You can also use TypeScript to type the props. 
+
+   ```astro title="src/components/MyCustomImageComponent.astro"
+    ---
+    import type { ImageMetadata } from "astro";
+    import { getImage } from "astro:assets";
+
+    interface Props {
+        mobileImgUrl: string | ImageMetadata;
+        desktopImgUrl: string | ImageMetadata;
+        alt: string;
+    }
+
+    const { mobileImgUrl, desktopImgUrl, alt } = Astro.props;
+    ---
+
+    ```
+
+3. Call the `getImage()` function and pass in the properties you want to use into the options object 
```

Comment:
```suggestion
3. Call the `getImage()` function and pass in the properties you want to use into the options object to define each separate image you want to create.
```
Something like this maybe? To emphasize we're doing this for each responsive image size?

I could also see something phrased in the opposite direction, like:

> Define each of your responsive images by calling the `getImage()` function with your desired properties.

Whatever you like!

Reply:
I like the second option 👍